web/src/stores/dandiset.ts

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable no-use-before-define */
+
+import { defineStore } from 'pinia';
+
+import axios from 'axios';
+import RefParser from '@apidevtools/json-schema-ref-parser';
+
+// eslint-disable-next-line import/no-cycle
+import { dandiRest } from '@/rest';
+import { User, Version } from '@/types';
+import { draftVersion } from '@/utils/constants';
+
+interface State {
+  dandiset: Version | null;
+  versions: Version[] | null,
+  loading: boolean,
+  owners: User[] | null,
+  schema: any,
+}
+
+export const useDandisetStore = defineStore('dandiset', {
+  state: (): State => ({
+    dandiset: null,
+    versions: null,
+    loading: false, // No mutation, as we don't want this mutated by the user
+    owners: null,
+    schema: null,
+  }),
+  getters: {
+    version: (state) => (state.dandiset ? state.dandiset.version : draftVersion),
+    schemaVersion: (state) => state.schema?.properties.schemaVersion.default,
+    userCanModifyDandiset: (state) => {
+      const user = dandiRest?.user as User | null;
+
+      // published versions are never editable, and logged out users can never edit a dandiset
+      if (state.dandiset?.metadata?.version !== draftVersion || !user) {
+        return false;
+      }
+      // if they're an admin, they can edit any dandiset
+      if (user.admin) {
+        return true;
+      }
+      // otherwise check if they are an owner
+      return !!(state.owners?.find((owner) => owner.username === user.username));
+    },
+  },
+  actions: {
+    async uninitializeDandisets() {
+      this.dandiset = null;
+      this.versions = null;
+      this.owners = null;
+      this.loading = false;
+    },
+    async initializeDandisets({ identifier, version }: Record<string, string>) {
+      this.uninitializeDandisets();
+
+      // this can be done concurrently, don't await
```

Comment:
Is this a TODO?

Reply:
No, I believe it's just there to clarify why there's no `await` in front of the following method so someone doesn't see it and think it's a bug. It was there in the old vuex code, I just copied it over.

Reply:
The confusion I had was between what you just said (explaining why the first statement has no `await`) and suggesting that someone should remove the following two `await`s and make all three operations concurrent.